### PR TITLE
ci: collect cluster state and pod logs on e2e test failure CAS-611

### PR DESCRIPTION
**Pull request by blaisedias**
Add scripts/e2e-cluster-dump.sh,
invoke from scripts/e2e-test.sh on test failures

### CODE REVIEW

File: scripts/e2e-cluster-dump.sh [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    fi
+}
+
+podlogs=1
+destdir=
+
+# Parse arguments
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+    -p|--path)
```
Comment (arne-rusek) [inline scripts/e2e-cluster-dump.sh:165]: For the sake of consistency across layers. Eases up orientation when reading if the same thing has the same name.
```suggestion
    -d|--destdir)
```

File: scripts/e2e-cluster-dump.sh [hunk 2]
```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
+
+help() {
+  cat <<EOF
+This script generates logs for mayastor pods and cluster state.
+
+Usage: $0 [OPTIONS]
+
+Options:
+  --path <path>   Location to store log files
```
Comment (arne-rusek) [inline scripts/e2e-cluster-dump.sh:10]: For the sake of consistency across layers. Eases up orientation when reading if the same thing has the same name.
```suggestion
  --destdir <path>   Location to store log files
```

File: scripts/e2e-cluster-dump.sh [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  --clusteronly   Only generate cluster information
+
+If --path is not specified the data is dumped to stdout
```
Comment (arne-rusek) [inline scripts/e2e-cluster-dump.sh:13]:
```suggestion
If --destdir is not specified the data is dumped to stdout
```

**Review by arne-rusek — APPROVED** [review]
LGTM, just some stylistic suggestions & `set -e`

File: scripts/e2e-cluster-dump.sh [hunk 4]
```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
+
```
[the remark this answers is not in the document]
Comment (blaisedias) [reply scripts/e2e-cluster-dump.sh:2]: Done

File: scripts/e2e-cluster-dump.sh [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    echo "#-- K8s Nodes -----------------------------"
+    kubectl describe nodes
+}
+
+function logs-csi-containers {
+    mayastor_csipods=$(kubectl -n mayastor get pods | grep mayastor-csi | sed -e 's/ .*//')
+    for pod in $mayastor_csipods
+    do
+        echo "# $pod csi-driver-registrar $* ---------------------------------"
+        kubectl -n mayastor logs "$@" "$pod"  csi-driver-registrar
```
[the remark this answers is not in the document]
Comment (blaisedias) [reply scripts/e2e-cluster-dump.sh]: Done

File: scripts/e2e-cluster-dump.sh (hunk 1 above)
Comment (blaisedias) [reply scripts/e2e-cluster-dump.sh]: Done

File: scripts/e2e-cluster-dump.sh (hunk 2 above)
Comment (blaisedias) [reply scripts/e2e-cluster-dump.sh]: Done

File: scripts/e2e-cluster-dump.sh (hunk 3 above)
Comment (blaisedias) [reply scripts/e2e-cluster-dump.sh]: Done

File: scripts/e2e-test.sh [hunk 6]
```diff
@@ -79,9 +79,17 @@ for dir in $TESTS; do
   fi
 done
 
+if [ -n "$test_failed" ]; then
+    # "$SCRIPTDIR"/e2e-cluster-dump.sh --path "/tmp/e2e-fail-logs"
```
Comment (paulyoong) [inline scripts/e2e-test.sh:83]: Should this be removed?

File: scripts/e2e-test.sh [hunk 7]
```diff
@@ -79,9 +79,17 @@ for dir in $TESTS; do
...
+    "$SCRIPTDIR"/e2e-cluster-dump.sh
+fi
+
 # must always run uninstall test in order to clean up the cluster
 cd "$SCRIPTDIR/../test/e2e/uninstall"
-go test
+if ! go test -v . -ginkgo.v -ginkgo.progress -timeout 0 ; then
+    "$SCRIPTDIR"/e2e-cluster-dump.sh --clusteronly
+    # "$SCRIPTDIR"/e2e-cluster-dump.sh --path /tmp/uninstall-fail-logs
```
Comment (paulyoong) [inline scripts/e2e-test.sh:91]: Should this be removed?

**Review by paulyoong — APPROVED** [review]

File: scripts/e2e-test.sh (hunk 6 above)
Comment (blaisedias) [reply scripts/e2e-test.sh]: Done

File: scripts/e2e-test.sh (hunk 7 above)
Comment (blaisedias) [reply scripts/e2e-test.sh]: Done

**Comment by blaisedias** [comment]
bors merge

**Pull request merged**